swarm/swarm.go

```diff
@@ -524,8 +524,33 @@ func (self *Swarm) SetChequebook(ctx context.Context) error {
 type Info struct {
 	*api.Config
 	*chequebook.Params
+	StoreStats *LocalStoreStats
 }
 
 func (self *Info) Info() *Info {
+	self.StoreStats.Info()
 	return self
 }
+
+type LocalStoreStats struct {
+	getters map[string]func() interface{}
+
+	DbCapacity   uint64
+	DbEntryCount uint64
+}
+
+func NewLocalStoreStats(l *storage.LocalStore) *LocalStoreStats {
+	getters := map[string]func() interface{}{}
+	getters["dbCapacity"] = l.DbStore.GetCapacity
+	getters["dbEntryCount"] = l.DbStore.GetEntryCount
+
+	stats := &LocalStoreStats{
+		getters: getters,
+	}
+	return stats
+}
+
+func (self *LocalStoreStats) Info() {
+	self.DbCapacity = self.getters["dbCapacity"]().(uint64)
+	self.DbEntryCount = self.getters["dbEntryCount"]().(uint64)
+}
```

Comment:
This seems a little complex, can it not just call the methods directly (i.e. `DbCapacity = localStore.GetCapacity()`)?

Also, why does it assign to `self` rather than returning a struct with the values in it?

Reply:
this is because of the general cramming of information under the `bzz` namespace. if i don't pass the getters to the `Info` type through the map - I'll have to start passing references to the `localStore` or `Swarm` and I really don't want to do that.
if this is a blocker then I'd really rather give the whole thing proper namespacing (and possibly also rename the `bzz` namespace to be `config` rather than `bzz` (referring to the `Service: &Info{...}` section)